ansible/reprovision-k8s-only.yaml

```diff
@@ -94,6 +94,9 @@
         hostname: "{{ api_server_elb.stdout }}"
       tags: k-prov
 
+    - pause:
+        prompt: "Check {{ stack_name }}-api.ft.com resolve to the correct LB ? (Enter to continue or Ctrl+C to abort)"
```

Comment:
From what we discussed yesterday, is it that we think it's more that the instances haven't been created yet, rather than the DNS record not updating?

We should update the message to check both the DNS record and the instance health (or just the instance health, if that's what we think the real problem is.

Reply:
We should not be worrying about instances not being created. Coz the cloudformation stack would fail if it doesn't get a success message from the instances it had created within 15M. And the ansible playbook would never proceed

Reply:
The CF stack gets created and exits successfully, which creates the ASGs.

However, the instances in the ASGs may not have come up yet or be attached to the ELB, and yesterday we found that this was why the provisioner broke on the next step - because there were no healthy API servers to handle requests.

Reply:
Perhaps it's not quite doing what we expect with the CF stack returning healthy? I didn't see it myself, it was @mihaylovmihail who saw the instances hadn't been created - he may have more info.

Reply:
It is interesting that we don't get this on every reprovisioning, which is why it feels like a timing issue to us.

Reply:
The CFs creates the ASG and EC2 instances. The CF will **never** succeed if the instances are unhealthy (a.k.a kubelet being up and it getting a success signal from the instance).

Reply:
I just forgot about the main reason for failing. Thank you for that note.

Reply:
I don't see how that could have happened tbh considering how cf works. This is the first time that somebody had reported it 🤷‍♀️

Reply:
Fair enough - can you two have a chat then and work out what *should* happen compared to what we saw yesterday? Perhaps I misunderstood when I thought the instances weren't ready. 🙂